accepted/00000-reboot-required-flag.md

```diff
@@ -10,50 +10,53 @@ The feature of showing in the Web UI when a system managed by Uyuni requires a r
 
-#### Salt Beacon
-For transactional systems (i.e. SLE Micro), our current approach for detecting and showing the reboot required indication is based on a Salt Beacon that runs each 10 seconds directly in the client system and notifies the server whenever the need for reboot changes. This strategy will remain for transactional systems, but it would also be possible to extend it for other Linux distributions.
+Additionally, a boolean indicating whether a reboot is required or not will be included in package profile update in order to be possible updating the information in case of master losing the beacon event for any reason.
 
-The main advantages of this strategy are:
- - Instantly detect the need for a reboot, regardless of what caused it.
- - Have a standardized way of checking for reboot required.
+##### Storing reboot information
 
-On the other hand, the main reason for not using this alternative is the risk of overloading the server with beacon events, especially for large installations. For example, applying a patch that makes a reboot to be required in multiple systems will prompt these client systems to transmit the beacon event in addition to package profile update, resulting in a doubling of the events that the server must process. Likewise, upon the client system reboot, the beacon will send the reboot required events just after the startup event.
+As the `reboot_info` beacon will be adapted to not fire events indicating that the reboot is not required, it is necessary to adapt the logic of the reboot required flag in server side. Currently there is a boolean database column `reboot_needed` at `suseMinionInfo` table for this purpose. This column will be modified to be a timestamp `reboot_required_after` and the server will need to compare this timestamp with the `uptime` of the system to determine if a reboot is required or not in a given moment.
 
+##### Changing Suse distros approach
 
+As it is possible to handle the reboot required information using the proposed approach, it will be used for all distributions. This implies changing the current approach for Suse systems to not query package/patch metadata for this purpose. Thereby, there will be a standardized implementation for the feature. Furthermore, in large installations, the query to list systems requiring reboot will have a better performance.
```

Comment:
```suggestion
As it is possible to handle the reboot required information using the proposed approach, it will be used for all distributions. This implies changing the current approach for SUSE systems to not query package/patch metadata for this purpose. Thereby, there will be a standardized implementation for the feature. Furthermore, in large installations, the query to list systems requiring reboot will have a better performance.
```